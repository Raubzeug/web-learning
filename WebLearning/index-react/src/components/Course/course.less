@import "../../styles/library.less";


.course {
    .flex-column;
    margin: 10px;
    padding: 10px;
    flex-basis: 300px;
    max-width: 400px;
    flex-grow: 1;

    &-meta {
        display: flex;
    }

    &-img {
        max-width: 100px;
        margin: 10px 0;
    }

    &-content {
        .flex-column;
        flex-grow: 1;
        
    }

    &-figure {
        text-align: center;
        padding-bottom: 10px;
        margin-top: auto;
        margin-bottom: auto;
    }

    &-figure p {
        text-align: left;
    }

    &-info-button {
        .non-decorated-link(@light-black-text);
        display: block;
        margin-top: auto;
        flex-shrink: 0;
        padding:5px 10px;
        text-align: center;
        font-size: 0.8em;
        font-weight:bold;
        background:white;
        border:none;
        background-image: @gradient-gray;
        box-shadow: @gray-button-shadow;
        text-shadow:0 1px 1px white;
        cursor: pointer;
    }
}
