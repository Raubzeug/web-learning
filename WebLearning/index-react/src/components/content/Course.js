import React from 'react';

const Course = ({course}) => (
    <article className="course">
        <h1 className="course-heading">{course.name}</h1>
        <div className="course-meta">
            <div className="course-category">
                <span className="course-icon course-tag">Категория: </span>
                <span>{course.lang}</span>
            </div>
        </div>
        <div className="course-content">
        <figure className="course-figure">
            <img className="course-img" src={require("../../images/" + course.img)} alt={course.lang} />
            <figcaption>
                <p>{course.caption}</p>
            </figcaption>
        </figure>
        <a className="course-info-button" href={course.url}>Информация о курсе</a>
        </div>
    </article>
)

export default Course