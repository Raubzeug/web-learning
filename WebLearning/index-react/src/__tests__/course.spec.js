import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Course from '../components/Course'

describe('course component', () => {
    test('renders properly', () => {
        const wrapper = shallow(<Course course={{
                    name: 'mockName',
                    lang: 'mockLang',
                    img: 'mockPic',
                    caption: 'mockCap',
                    url: 'mockUrl'
                }}/>)
        const heading = wrapper.find('.course-heading').first().text()
        const link = wrapper.find('.course-info-button').first().prop('href')
        expect(shallowToJson(wrapper)).toMatchSnapshot()
        expect(heading).toEqual('mockName')
        expect(link).toContain('mockUrl')
    })
})