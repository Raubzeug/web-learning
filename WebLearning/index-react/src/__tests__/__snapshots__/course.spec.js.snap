// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`course component renders properly 1`] = `
<article
  className="course"
>
  <h1
    className="course-heading"
  >
    mockName
  </h1>
  <div
    className="course-meta"
  >
    <div
      className="course-category"
    >
      <span
        className="course-icon course-tag"
      >
        Категория: 
      </span>
      <span>
        mockLang
      </span>
    </div>
  </div>
  <div
    className="course-content"
  >
    <figure
      className="course-figure"
    >
      <img
        alt="mockLang"
        className="course-img"
        src="mockPic"
      />
      <figcaption>
        <p>
          mockCap
        </p>
      </figcaption>
    </figure>
    <a
      className="course-info-button"
      href="mockUrl"
    >
      Информация о курсе
    </a>
  </div>
</article>
`;
